### PR TITLE
Add pipeline files for matrix-appservice-slack

### PR DESCRIPTION
**Pull request by Half-Shot**
This was stored in the community repo and therefore meant we couldn't run buildkite for community PRs.

### CODE REVIEW

File: matrix-appservice-slack/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  - label: ":nodejs: 14 :postgres: 11 :mocha: Postgres Test"
+    command:
+      - "npm install"
+      - "npm run build"
+      - "npm run test:postgres"
+    plugins:
+      - docker-compose#v2.1.0:
+          run: testenv
+          config:
+            - .buildkite/docker-compose.postgres.yml
```
Comment (richvdh) [inline matrix-appservice-slack/pipeline.yml]: you'll need to download this before it will work

File: matrix-appservice-slack/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+steps:
+  - label: ":eslint: Lint"
+    command:
+      - "npm install"
+      - "npm run lint"
+    plugins:
+      - docker#v3.0.1:
+          image: "node:14"
```
Comment (richvdh) [inline matrix-appservice-slack/pipeline.yml]: needs more `mount-buildkite-agent`

**Review by richvdh — CHANGES_REQUESTED** [review]
could you have a look at https://gitlab.matrix.org/new-vector/internal/-/wikis/BuildKite please

**Comment by Half-Shot** [comment]
Oops. thanks. Should be good now?

File: matrix-appservice-slack/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
```
Comment (richvdh) [inline matrix-appservice-slack/pipeline.yml:9]: still needs more mount-buildkite-agent

**Review by richvdh — CHANGES_REQUESTED** [review]

**Comment by Half-Shot** [comment]
That's about as much mount-buildkite-agent as I can do. I wish it was off by default :(

File: matrix-appservice-slack/docker-compose.postgres.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  testenv:
+    image: node:14
+    depends_on:
+      - postgres
+    environment:
+      SLACKBRIDGE_TEST_PGDB: "slackintegtest"
+      SLACKBRIDGE_TEST_PGURL: "postgresql://postgres:postgres@postgres"
+    working_dir: /app
+    volumes:
+      - ..:/app
```
Comment (richvdh) [inline matrix-appservice-slack/docker-compose.postgres.yml]: what is this `..` relative to? mounting the parent of the build directory sounds like a dangerous idea.

Comment (Half-Shot) [reply matrix-appservice-slack/docker-compose.postgres.yml]: Hm, I had changed that. Anyway, commited the correct mapping.

**Review by richvdh — APPROVED** [review]
woot

**Pull request merged**